framework/core/command_buffer.h

```diff
@@ -124,33 +128,25 @@ class CommandBuffer
 	void set_specialization_constant(uint32_t constant_id, const std::vector<uint8_t> &data);
 
 	/**
-	 * @brief Stores additional data which is prepended to the
-	 *        values passed to the push_constants_accumulated() function
-	 * @param data Data to be stored
+	 * @brief Records byte data into the command buffer to be pushed as push constants to each draw call
+	 * @param values The byte data to store
 	 */
-	template <class T>
-	void set_push_constants(const T &data);
-
-	void set_push_constants(const std::vector<uint8_t> &values);
-
-	void push_constants_accumulated(const std::vector<uint8_t> &values, uint32_t offset = 0);
+	void push_constants(const std::vector<uint8_t> &values);
 
 	template <typename T>
-	void push_constants_accumulated(const T &value, uint32_t offset = 0)
+	void push_constants(const T &value)
 	{
-		push_constants_accumulated(std::vector<uint8_t>{reinterpret_cast<const uint8_t *>(&value),
-		                                                reinterpret_cast<const uint8_t *>(&value) + sizeof(T)},
-		                           offset);
-	}
+		auto data = to_bytes(value);
 
-	void push_constants(uint32_t offset, const std::vector<uint8_t> &values);
+		uint32_t size = to_u32(stored_push_constants.size() + data.size());
 
-	template <typename T>
-	void push_constants(uint32_t offset, const T &value)
-	{
-		push_constants(offset,
-		               std::vector<uint8_t>{reinterpret_cast<const uint8_t *>(&value),
-		                                    reinterpret_cast<const uint8_t *>(&value) + sizeof(T)});
+		if (size > max_push_constants_size)
+		{
+			LOGE("Push constant limit exceeded ({} / {} bytes)", size, max_push_constants_size);
+			throw std::runtime_error("Cannot overflow push constant limit");
+		}
+
+		push_constants(to_bytes(value));
```

Comment:
Is there not a more efficient way to get the data to push_constants() without making an intermediate vector?